internal/pkg/testproject/provide.go

```diff
@@ -0,0 +1,90 @@
+// nolint forbidigo
+package testproject
+
+import (
+	"fmt"
+	"github.com/keboola/keboola-as-code/internal/pkg/env"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/spf13/cast"
+)
+
+var projects []*Project      // nolint gochecknoglobals
+var initLock = &sync.Mutex{} // nolint gochecknoglobals
+
+func GetTestProject(t *testing.T, envs *env.Map) *Project {
+	t.Helper()
+	initProjects()
+
+	if len(projects) == 0 {
+		panic(fmt.Errorf(`no test project`))
+	}
+
+	for {
+		// Try to find a free project
+		for _, p := range projects {
+			if p.tryLock(t, envs) {
+				return p
+			}
+		}
+
+		// No free project -> wait
+		time.Sleep(50 * time.Millisecond)
```

Comment:
Tu je `testproject.GetTestProject` vrati volny test project alebo caka, kym sa nejaky neuvolni.
Kedze je pouzity zamok `fslock`, tak tato metoda moze byt bez problemov spustana z viacerych procesov.

Mozes si aj celu test suite spusti naraz viac krat, a nenastane kolizia.

Reply:
V tomto PR sa to spusta stale oproti jednemu projektu v CI, v dalsich PR pridavam viacero projektov (aby tu toho nebolo vela).